src/lib.rs

```diff
@@ -97,10 +98,11 @@ async fn handle(
                 .map(ToOwned::to_owned)
                 .unwrap_or_else(|| PathAndQuery::from_static(""));
 
+            let split_uri: Vec<&str> = market_uri.split("://").collect();
```

Comment:
Looks good and could be improved, but we can make it much simpler.
We can concat the market_url and path_and_query. We can do that with `format!("{}{}", market_url, path_and_query)` and we can use this to parse it into a URI: https://docs.rs/http/0.1.17/http/uri/struct.Uri.html#examples
`.parse::<Uri>()?`